intellij-ext/src/main/kotlin/com/metalbear/mirrord/MirrordEnabler.kt

```diff
@@ -35,10 +47,58 @@ class MirrordEnabler : ToggleAction() {
     override fun setSelected(e: AnActionEvent, state: Boolean) {
         if (state) {
             notify("mirrord enabled", NotificationType.INFORMATION, e.project)
+            e.project?.let { checkVersion(it) }
         } else {
             notify("mirrord disabled", NotificationType.INFORMATION, e.project)
         }
 
         MirrordListener.enabled = state
     }
+
+    private val pluginId = PluginId.getId("com.metalbear.mirrord")
+    private val version: String? = PluginManagerCore.getPlugin(pluginId)?.version
+    private val versionCheckEndpoint: String =
+        "https://version.mirrord.dev/get-latest-version?source=3&version=$version"
+    private var versionCheckDisabled
+        get() = PropertiesComponent.getInstance().getBoolean("versionCheckDisabled", false)
+        set(value) {
+            PropertiesComponent.getInstance().setValue("versionCheckDisabled", value)
+        }
+
+    private fun checkVersion(project: Project) {
```

Comment:
Could you add some documentation to what this function does? From a quick glance it looks like it does more than just version checking?

Reply:
Added.